_api-reference/ingest-apis/ip2geo.md

```diff
@@ -0,0 +1,212 @@
+
+#### Example: Request
+
+```
+GET /_plugins/geospatial/ip2geo/datasource/my-datasource
+```
+
+#### Example: Response
+
+```
+{
+  "datasources": [
+    {
+      "name": "my-datasource",
+      "state": "AVAILABLE",
+      "endpoint": "https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json",
+      "update_interval_in_days": 3,
+      "next_update_at_in_epoch_millis": 1685125612373,
+      "database": {
+        "provider": "maxmind",
+        "sha256_hash": "0SmTZgtTRjWa5lXR+XFCqrZcT495jL5XUcJlpMj0uEA=",
+        "updated_at_in_epoch_millis": 1684429230000,
+        "valid_for_in_days": 30,
+        "fields": [
+          "country_iso_code",
+          "country_name",
+          "continent_name",
+          "region_iso_code",
+          "region_name",
+          "city_name",
+          "time_zone",
+          "location"
+        ]
+      },
+      "update_stats": {
+        "last_succeeded_at_in_epoch_millis": 1684866730192,
+        "last_processing_time_in_millis": 317640,
+        "last_failed_at_in_epoch_millis": 1684866730492,
+        "last_skipped_at_in_epoch_millis": 1684866730292
+      }
+    }
+  ]
+}
+```
+
+## Updating IP2geo datasource
+
+Update IP2geo datasource. The GeoIP data from the new endpoint should contain all fields in GeoIP data from the current endpoint for the update to succeed.
+
+#### Example: Request
+
+```json
+{
+    "endpoint": https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json,
+    "update_interval_in_days": 10
+}
+```
+
+#### Example: Response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
+## Creating the IP2geo processor
+
+
+
+#### Example: Request
+
+```json
+{
+   "description":"convert ip to geo",
+   "processors":[
+    {
+        "ip2geo":{
+            "field":"ip",
+            "datasource":"my-datasource"
+        }
+    }
+   ] 
+}
+
+#### Example: Response
+
+```json
+{
+	"acknowledged": true
+}
+```
+
+## Using the IP2geo processor in a pipeline
+
+The following table describes `ip2geo` options
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| field | yes | - | The field to get the ip address from for the geographical lookup. |
+| target_field | no | ip2geo | The field that will hold the geographical information looked up from the Maxmind database. |
+| database | no | geoip.maps.opensearch.org/v1/geolite2-city/manifest.json | The database filename referring to a database the module ships with or a custom database in the ingest-geoip config directory. |
+| properties | no |  Default field depends on what is available in `database`. | Controls what properties are added to the target_field based on the geoip lookup. |
+| ignore_missing | no | false | If `true` and `field` does not exist, the processor quietly exits without modifying the document. |
+| first_only | no | true | If `true` only first found geoip data will be returned, even if field contains array. |
+
+
```

Comment:
Add below contents here

--
### Node Settings
The IP2geo datasource and IP2geo processor has node settings

IP2Geo datasource:

* plugins.geospatial.ip2geo.datasource.endpoint
Default endpoint for create datasource API. Defaults to https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json
* plugins.geospatial.ip2geo.datasource.update_interval_in_days
Default update interval for create datasource API. Defaults to 3.
* plugins.geospatial.ip2geo.datasource.indexing_bulk_size
Max number of documents to ingest in [bulk request](https://opensearch.org/docs/1.2/opensearch/rest-api/document-apis/bulk/) during IP2geo datasource creation process. Defaults to 10,000.

IP2Geo processor:

* plugins.geospatial.ip2geo.processor.max_bundle_size
Maximum number of searches to include in a [multi-search request](https://opensearch.org/docs/latest/api-reference/multi-search/) when enriching documents. Defaults to 100.
* plugins.geospatial.ip2geo.processor.max_concurrent_searches
Maximum number of concurrent [multi-search requests](https://opensearch.org/docs/latest/api-reference/multi-search/) to run when enriching documents. The default depends on your node count and search thread pool size. Higher values can improve performance, but risk overloading the cluster.